src/main/java/io/jenkins/plugins/generic/event/HttpEventSender.java

```diff
@@ -25,36 +26,69 @@
 public class HttpEventSender implements EventSender {
 
     private final Logger logger = Logger.getLogger(HttpEventSender.class.getName());
+    private final CustomCloseableHttpAsyncClient myClient = new CustomCloseableHttpAsyncClient();
 
     @Override
     public void send(Event event) {
         EventDataTransformers.INSTANCE.transform(event);
 
-        // serialize event to JSON string
-        String receiver = EventGlobalConfiguration.get().getReceiver();
+        final String receiver = EventGlobalConfiguration.get().getReceiver();
+
         if (StringUtils.isBlank(receiver)) {
             logger.info("Skipped event sending due to receiver URL not set");
             return;
         }
-        // TODO Make HTTP request asynchronous
-        try (final CloseableHttpClient client = HttpClients.createDefault()) {
-            String eventJSON = JSONObject.fromObject(event, new EventJsonConfig()).toString(4);
-            HttpEntity httpEntity = EntityBuilder.create()
-                    .setContentType(ContentType.APPLICATION_JSON)
-                    .setText(eventJSON)
-                    .build();
-            HttpPost httpPost = new HttpPost(receiver);
-            httpPost.setEntity(httpEntity);
-            try (final CloseableHttpResponse response = client.execute(httpPost)) {
-                HttpEntity responseEntity = response.getEntity();
-                if (logger.isLoggable(Level.INFO)) {
-                    logger.info("Event send successfully, and response: " + EntityUtils.toString(responseEntity));
+
+        try {
+            new Thread(() -> myClient.sendPost(receiver, event, new FutureCallback<SimpleHttpResponse>() {
+                @Override
+                public void completed(SimpleHttpResponse response) {
+                    logger.info("Event send succeeded, response: " + response.getBodyText());
                 }
-                EntityUtils.consumeQuietly(responseEntity);
-            }
+
+                @Override
+                public void failed(Exception e) {
+                    logger.info("Event send has been failed, response: " + e);
+                }
+
+                @Override
+                public void cancelled() {
+                    logger.info("Event send has been cancelled");
+                }
+            })).start();
         } catch (Exception e) {
             logger.log(Level.WARNING, "Failed to send event to " + receiver, e);
         }
+
     }
 
+    static public class CustomCloseableHttpAsyncClient {
+        private final CloseableHttpAsyncClient httpClient;
```

Comment:
Please make this client as a static field here. It's expensive to initialize the CloseableHttpAsyncClient at every initialization of CustomCloseableHttpAsyncClient.